src/gui/dialogs/edit_text.cpp

```diff
@@ -41,14 +46,37 @@ namespace dialogs
 
 REGISTER_DIALOG(edit_text)
 
-//TODO: add  a way to disallow certain characters (like spaces or ")
 edit_text::edit_text(const std::string& title,
 					   const std::string& label,
-					   std::string& text)
+					   std::string& text,
+					   bool disallow_empty)
+   : disallow_empty_(disallow_empty)
 {
 	register_label("title", true, title, true);
 	register_label("label", true, label, true);
 	register_text("text", true, text, true);
 }
+
+void edit_text::pre_show(window& window)
+{
+   if(disallow_empty_) {
+      text_box& text = find_widget<text_box>(&window, "text", false);
+      connect_signal_notify_modified(text, std::bind(&edit_text::on_text_change, this));
+      on_text_change();
+   }
+}
+
+void edit_text::on_text_change()
+{
+   text_box& text = find_widget<text_box>(get_window(), "text", false);
+   button& ok_button = find_widget<button>(get_window(), "ok", false);
+
+   auto clean_value = text.get_value();
+	boost::trim(clean_value);
```

Comment:
You don't need to create a temporary here. You can use `boost::trim_copy`.

Reply:
The temporary is still created, though it is no longer used.